cfgov/unprocessed/apps/youth-employment-success/js/data-types/notifications.js

```diff
@@ -0,0 +1,74 @@
+const ALERT_TYPES = Object.freeze( {
+  NONE: 'NONE',
+  INVALID: 'INVALID',
+  IN_BUDGET: 'IN_BUDGET',
+  OUT_OF_BUDGET: 'OUT_OF_BUDGET',
+  HAS_TODOS: 'HAS_TODOS'
+} );
+
+/**
+ * Bitmask to resolve which alert should be shown in the
+ * `route-details` section, based on the current state of the
+ * form.
+ *
+ * Using a bitmask here has some advantages:
+ *
+ * 1). It reduces the need for messy if/else logic.
+ * 2). It enables new flags to be added easily when alert logic requirements change.
+ * 3). It provides basic state-machine / rules engine-like behavior
+ *     without needing a lot of code.
+ * 4). It ensures that no matter what the state of the application, there will
+ *     be at most one displayed alert, and which alert is shown won't need to be resolved
+ *     heirarchically.
+ * 5). It leaves the final implementation of what alerts should be shown
+ *     up to the view making use of the bitmask.
+ *
+ * The bitmask's values are each a power of two. Left-shifting
+ * 1 by an increasing right-hand operand handles this automatically
+ * by resolving to the following formula:
+ *
+ * Where x is the left-hand operand and n is the right:
+ *
+ * x * (2 ^ n), or e.g. 1 * (2 ^ 3) -> 8.
+ *
+ */
+const ALERT_BITMASK_ENUM = Object.freeze( {
```

Comment:
Couldn't you just set the values directly, like:
```
const ALERT_BITMASK_ENUM = Object.freeze( {
  [ALERT_TYPES.NONE]: 0,
  [ALERT_TYPES.INVALID]: 1,
  [ALERT_TYPES.IN_BUDGET]: 2,
  [ALERT_TYPES.OUT_OF_BUDGET]: 4,
  [ALERT_TYPES.HAS_TODOS]: 8
} );
```

Or are the shift operations to make it clearer this is used with bitwise operations?

Reply:
That, and the shift operators do it for you without specifying manually, which I always find is prone to error (especially on my part) 😄

Reply:
Possibly remove the `<< 0` on `ALERT_TYPES.INVALID`, since `ALERT_TYPES.NONE` doesn't have a shift either, and then you're only using the shift operators that actually do something.

Reply:
That makes sense to me!